core/src/main/java/org/springframework/security/core/annotation/AuthenticationPrincipal.java

```diff
@@ -46,4 +46,33 @@
 	 * @return
 	 */
 	boolean errorOnInvalidType() default false;
+
+	/**
+	 * If specified will use the provided SpEL expression to resolve the principal. This
+	 * is convenient if users need to transform the result.
+	 *
+	 * <p>
+	 * For example, perhaps the user wants to resolve a CustomUser object that is final
+	 * and is leveraging a UserDetailsService. This can be handled by returning an object
+	 * that looks like:
+	 * </p>
+	 *
+	 * <code>
```

Comment:
Javadoc is not rendering properly...
![screen shot 2016-05-03 at 5 37 14 pm](https://cloud.githubusercontent.com/assets/10884212/14999257/198e593c-1156-11e6-97d8-b0dd4e22f992.png)